### PR TITLE
OADP-605: Fix edge case for already exists resources

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**

### CODE REVIEW

File: pkg/restore/restore.go [hunk 1]
```diff
@@ -1249,7 +1249,21 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
...
+	objectExists := false
+	if !isAlreadyExistsError {
+		_, err := resourceClient.Get(obj.GetName(), metav1.GetOptions{})
+		if !apierrors.IsNotFound(err) {
+			objectExists = true
+		}
+		if err != nil && !apierrors.IsNotFound(err) {
+			errs.Add(namespace, err)
+			return warnings, errs
+		}
```
Comment (kaovilai) [inline pkg/restore/restore.go]: This reduce call to !apierrors.IsNotFound(err). More performance?
```suggestion
		if !apierrors.IsNotFound(err) {
			objectExists = true
		}
		if err != nil && objectExists {
			errs.Add(namespace, err)
			return warnings, errs
		}
```

**Comment by kaovilai** [comment]
Do we want to scope this edge case to if `restoreErr` contain regex `webhook.*denied the request` ?
https://github.com/openshift/velero/blob/61847a78e3eb83c3676505568ce289ea3d26d365/pkg/restore/restore.go#L1246

**Comment by kaovilai** [comment]
```go
deniedBy := fmt.Sprintf("admission webhook %q denied the request", webhookName)
```
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/errors/statuserror.go#L29

**Comment by shubham-pampattiwar** [comment]
@kaovilai We want to keep this generic to work for general errors that we may not foresee in the future.

**Review by kaovilai — APPROVED** [review]

**Comment by kaovilai** [comment]
We might wanna compare specs here.. and add warning if spec is different than "in-cluster version"

File: pkg/restore/restore.go [hunk 2]
```diff
@@ -1249,7 +1249,21 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
...
+			objectExists = true
+		}
+		if err != nil && objectExists {
+			errs.Add(namespace, err)
+			return warnings, errs
+		}
+	}
+
+	if isAlreadyExistsError || objectExists {
 		fromCluster, err := resourceClient.Get(name, metav1.GetOptions{})
```
Comment (kaovilai) [inline pkg/restore/restore.go]: Is it possible to reuse the get call from here?

**Comment by shubham-pampattiwar** [comment]
@kaovilai the spec comparison and subsequent processing is done here https://github.com/openshift/velero/blob/konveyor-dev/pkg/restore/restore.go#L1273

File: pkg/restore/restore.go (hunk 2 above)
Comment (sseago) [reply pkg/restore/restore.go]: @kaovilai We could declare the var `fromCluster` where we define objectExsts, and set it in the new Get call above, then update this Get call to only run if fromCluster is nil.

**Review by sseago — APPROVED** [review]
Looks good, but shouldn't we push this upstream first? Or is this needed today because it's a blocker for 1.1? In any case, as soon as it's approved/merged upstream, we want to use that commit and not the local one. Also, konveyor-1.9.0 is probably the wrong branch here. I just created konveyor-1.9.1 yesterday, and that's the branch we'll be pulling oadp-1.1 from when those branches are cut.

**Comment by sseago** [comment]
Also see the suggestion above for reducing the number of Get calls.

**Comment by shubham-pampattiwar** [comment]
Closing this one, opened another one for konveyor-1.9 https://github.com/openshift/velero/pull/186

**Pull request closed without merging**